packages/labs/src/components/popover/popoverMigrations.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import { Position } from "@blueprintjs/core";
+import { Placement } from "./popover2";
+
+/**
+ * Get the first non-undefined prop value, falling back to the specified default value.
+ * @param candidates the ordered list of value candidates
+ * @param defaultValue the fallback value
+ */
+export function resolvePropValue<T>(candidates: Array<T | undefined>, defaultValue: T): T {
```

Comment:
How about `(defaultValue, ...candidates)`? Won't need to clone the array before reversing.